kivy/core/window/__init__.py

```diff
@@ -933,6 +933,39 @@ def _set_top(self, value):
     the position set in :class:`~kivy.config.Config`.
     '''
 
+    def _get_opacity(self):
+        return self._get_window_opacity()
+
+    def _set_opacity(self, opacity):
+        if opacity > 0.0 and opacity < 1.0:
```

Comment:
SDL2 will clamp the value if less than 0 or greater than 1, so I would remove range testing from this method (and logger warning) and let specific window implementation decide on what to do. Since SDL2 will clamp there is no need to range test in `WindowSDL`, but just test for errors https://github.com/kivy/kivy/pull/8296/files#r1248874160

Reply:
@misl6 Do you agree?

Reply:
@pythonic64 thanks for chiming in! ❤️

I partially agree:
- Testing for errors (and logging it) via `SDL_GetError` is absolutely something we should have. (As for any reason, even if the value is in the supported range, something bad may happen, and the user/developer needs to be informed)

- Defining a supported range on `WindowBase` looks like a good approach instead, as will force us in the future to standardize the opacity range along all the window providers. (the provider-specific implementation will take care of translating the `0.0 - 1.0` range to a supported value). By not defining a supported range, the risk is that in the future we may introduce a window provider with `0-100` range, which is incompatible with the `0.0-1.0` range.
By defining a supported range, Kivy is responsible to maintain compatibility by applying the needed translations.

Reply:
> I agree on standardizing opacity range to 0.0 - 1.0 but imho that should be (and in this pull request already is) stated only in the docs. Similarly, we don't do range test for each component in ColorProperty, but we do state supported range in the docs. If in the future we have another window provider we will still respect the defined range from the docs.

Yep. Makes sense!

@pythonic64 and @mak8kammerer feel free to ping me when you need a final review or to run tests (as @mak8kammerer is a new contributor, and workflow runs need to be manually approved)

Reply:
@mak8kammerer SDL will clamp the value, that's stated in the docs, "Remarks" section: https://wiki.libsdl.org/SDL2/SDL_SetWindowOpacity#remarks .

In our discussion above we agree that range should be specified in the docs only as we don't expect users assigning values outside of that range. Since SDL is clamping the value that's fine, but if in the future we have another window provider which doesn't clamp the value but raises an error then we will decide in that window specific implementation to either:
1. clamp and set the clamped value
2. catch and log the error, leaving the current opacity value unchanged

`Widget.opacity` is an another example where we don't do range check and assigning values outside of `[0.0-1.0]` does not raise an error.